packages/blaze-query/src/hydra.ts

```diff
@@ -0,0 +1,758 @@
+  hydraHeadId: string;
+  snapshotUtxo: { [txRef: string]: TxOut };
+  timestamp: string;
+  hydraNodeVersion: string;
+}
+
+export interface PeerConnected {
+  tag: "PeerConnected";
+  peer: string;
+  seq: number;
+  timestamp: string;
+}
+
+export interface PeerDisconnected {
+  tag: "PeerDisconnected";
+  peer: string;
+  seq: number;
+  timestamp: string;
+}
+
+export interface PeerHandshakeFailure {
+  tag: "PeerHandshakeFailure";
+  remoteHost:
+    | {
+        tag: "IPv4";
+        ipv4: string;
+      }
+    | {
+        tag: "IPv6";
+        ipv6: string;
+      };
+  ourVersion: number;
+  theirVersions: number[];
+  seq: number;
+  timestamp: string;
+}
+
+export interface HeadIsInitializing {
+  tag: "HeadIsInitializing";
+  headId: string;
+  parties: Party[];
+  seq: number;
+  timestamp: string;
+}
+
+export interface Committed {
+  tag: "Committed";
+  parties: Party[];
+  utxo: { [txRef: string]: TxOut };
+  seq: number;
+  timestamp: string;
+}
+
+export interface HeadIsOpen {
+  tag: "HeadIsOpen";
+  headId: string;
+  utxo: { [txRef: string]: TxOut };
+  seq: number;
+  timestamp: string;
+}
+
+export interface HeadIsClosed {
+  tag: "HeadIsClosed";
+  headId: string;
+  snapshotNumber: number;
+  contestationDeadline: string;
+  seq: number;
+  timestamp: string;
+}
+
+export interface HeadIsContested {
+  tag: "HeadIsContested";
+  headId: string;
+  snapshotNumber: number;
+  contestationDeadline: string;
+  seq: number;
+  timestamp: string;
+}
+
+export interface ReadyToFanout {
+  tag: "ReadyToFanout";
+  headId: string;
+  seq: number;
+  timestamp: string;
+}
+
+export interface HeadIsAborted {
+  tag: "HeadIsAborted";
+  headId: string;
+  utxo: { [txRef: string]: TxOut };
+  seq: number;
+  timestamp: string;
+}
+
+export interface HeadIsFinalized {
+  tag: "HeadIsFinalized";
+  headId: string;
+  utxo: { [txRef: string]: TxOut };
+  seq: number;
+  timestamp: string;
+}
+
+export interface TxValid {
+  tag: "TxValid";
+  headId: string;
+  transaction: Transaction;
+  seq: number;
+  timestamp: string;
+}
+
+export interface TxInvalid {
+  tag: "TxInvalid";
+  headId: string;
+  utxo: { [txRef: string]: TxOut };
+  transaction: Transaction;
+  validationError: { reason: string };
+  seq: number;
+  timestamp: string;
+}
+
+export interface SnapshotConfirmed {
+  tag: "SnapshotConfirmed";
+  headId: string;
+  snapshot: Snapshot;
+  seq: number;
+  timestamp: string;
+}
+
+export interface GetUTxOResponse {
+  tag: "GetUTxOResponse";
+  headId: string;
+  utxo: { [txRef: string]: TxOut };
+  seq: number;
+  timestamp: string;
+}
+
+export interface InvalidInput {
+  tag: "InvalidInput";
+  reason: string;
+  input: string;
+  seq: number;
+  timestamp: string;
+}
+
+export interface PostTxOnChainFailed {
+  tag: "PostTxOnChainFailed";
+  postChainTx:
+    | InitTx
+    | AbortTx
+    | CollectComTx
+    | DecrementTx
+    | CloseTx
+    | ContestTx
+    | FanoutTx;
```

Comment:
Not sure if it's worthwhile to properly type `postChainTx`. From an API client's point of view, these are very much "internal server errors" and keeping them in an unparsed / untyped form is likely okay and breaks less in case we fix the abstraction leak.